lib/order_optimizer.rb

```diff
@@ -56,7 +56,7 @@ def count_and_remainder_for_sku(quantity, sku)
     count, remainder = quantity.divmod(sku.quantity)
 
     if sku.min_quantity && count * sku.quantity < sku.min_quantity
-      new_count = (sku.min_quantity.to_f / sku.quantity).ceil
+      new_count = (sku.min_quantity.to_f / sku.quantity.to_f).ceil
```

Comment:
As @Meo404 pointed out, this needs to handle both Integer/Integer and BigDecimal/BigDecimal and the other combinations of those. The problem here is really floats. Converting our lovely trustworthy inputs into Floats suddenly makes them less reliable. I'm going to investigate if there's an option where we don't do this conversion at all.